algorithms/kernel/svm/svm_train_boser_impl.i

```diff
@@ -498,16 +500,22 @@ SVMTrainTask<algorithmFPType, ParameterType, cpu>::~SVMTrainTask()
  * \param[in] C     Upper bound in constraints of the quadratic optimization problem
  */
 template <typename algorithmFPType, typename ParameterType, CpuType cpu>
-services::Status SVMTrainTask<algorithmFPType, ParameterType, cpu>::init(algorithmFPType C)
+services::Status SVMTrainTask<algorithmFPType, ParameterType, cpu>::init(algorithmFPType C, const NumericTablePtr & wTable)
 {
     DAAL_ITTNOTIFY_SCOPED_TASK(init);
 
-    algorithmFPType * grad    = _grad.get();
-    const algorithmFPType * y = _y.get();
+    ReadColumns<algorithmFPType, cpu> mtW(wTable.get(), 0, 0, _nVectors);
+    DAAL_CHECK_BLOCK_STATUS(mtW);
+    const algorithmFPType * weights = mtW.get();
+
+    algorithmFPType * const grad    = _grad.get();
+    const algorithmFPType * const y = _y.get();
+    algorithmFPType * const cw      = _cw.get();
     for (size_t i = 0; i < _nVectors; i++)
     {
         grad[i] = -y[i];
-        updateI(C, i);
+        cw[i]   = weights ? weights[i] * C : C;
```

Comment:
seems it would not affect performance significantly, but we could consider introduction of bool template parameter `hasWeights` to `init(...)` to avoid this condition in compile time.

Reply:
The introduction of the template will not change the check in runtime. It really does not affect performance.